Middleware/GuzzleRequestSpanning.php

```diff
@@ -30,11 +32,39 @@ public function __invoke(callable $handler): callable
             $promise = $handler($request, $options);
             return $promise->then(
                 function (ResponseInterface $response) {
-                    $this->requestSpanning->finish($response->getStatusCode());
-                    $this->tracing->finishActiveSpan();
-                    return $response;
+                    return $this->onFulfilled($response);
```

Comment:
Just an idea: too keep the focus within this layer, we could also make `onFulfilled()` a `void` function (like `onRejected()`), and `return $response;`.

This way, the developer does not need to dive into `onFulfilled()` to grasp, what is returned from this function.

Reply:
done